### ISSUE TITLE
Add Architecture option to "docker images" list

### ISSUE THREAD

**Issue by ericsmalling** [docker/cli#3060]
**Description**
Please add the ability to see the architecture for images in the cache via `docker images` / `docker image ls`.

Even if it's just an additional `--format` field, that would be valuable

**Comment by BretFisher** [comment]
I agree with this! On an arm device, like a Apple M1, it's confusing what docker is picking for arm vs. amd64 images. I just had a case where I was accidentally using a amd64 image on arm64 because I had previously pulled that specific platform and forgot it was different... then docker just defaults to your last pulled platform for that tag. 

To me, `platform` is just part of the image name (repo name + tag + platform), and it should be viewable without needing a inspect command.

Bonus points if there's an asterisk if the platform of image <> platform of dockerd, to highlight where you've manually chosen to emulate.

**Comment by sam-thibault** [comment]
see also:
- https://github.com/moby/moby/issues/44582
